### PR TITLE
Feature/jr2 766 Calendar work with keyboard

### PR DESCRIPTION
**Pull request by Invizible**

### CODE REVIEW

File: src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts [hunk 1]
```diff
@@ -61,6 +63,12 @@ export class JobPublicationToolComponent implements OnInit, OnDestroy {
     @Input()
     userData: UserData;
 
+    @ViewChild('publicationStartDateEl')
+    publicationStartDateEl: ElementRef;
```
Comment (michael-wirth) [inline src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts]: what does the suffix `El` mean? is it for "ElementRef" maybe we should use not abbreviate the term to make it more clear. @birom what do you think? Do we have a naming convention for ElementRef? fields?

File: src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts [hunk 2]
```diff
@@ -82,6 +90,23 @@ export class JobPublicationToolComponent implements OnInit, OnDestroy {
 
     private unsubscribe$ = new Subject<void>();
 
+    private static ngbDateStructValidator(control: AbstractControl): ValidationErrors | null {
```
Comment (michael-wirth) [inline src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts]: what is this validator doing? what means `ngb`? Can we find a better name?

Comment (michael-wirth) [reply src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts]: Also describe the what the Validator is doing. Also provide valid and invalid formats as example comment

Comment (Invizible) [reply src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts]: I figured, we do not need this validator, will remove

File: src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts (hunk 1 above)
Comment (birom) [reply src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts]: No, we do not really have.

File: src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.html [hunk 3]
```diff
@@ -151,6 +152,10 @@
                                                    aria-hidden="true"></i>
                                             </button>
                                         </div>
+                                        <jr2-input-error-message [formGroup]="jobPublicationForm" path="job.publicationStartDate.date"
+                                                                 [errorCodes]="['required', 'ngbDate.invalid']"
```
Comment (birom) [inline src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.html]: Error code 'ngbDate.requiredBefore' is missing here.

File: src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.html [hunk 4]
```diff
@@ -182,20 +187,25 @@
...
+                                                   [minDate]="publicationEndDateMin"
+                                                   (focus)="endDatePicker.close()"/>
                                             <button class="input-group-addon"
                                                     (click)="endDatePicker.toggle(); startDatePicker.close()"
                                                     type="button"
                                                     [disabled]="publicationEndDateIsPermanent">
                                                 <i class="fa fa-calendar-o" aria-hidden="true"></i>
                                             </button>
                                         </div>
+                                        <jr2-input-error-message [formGroup]="jobPublicationForm" path="job.publicationEndDate.date"
```
Comment (birom) [inline src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.html]: Error code 'ngbDate.requiredBefore' is missing here.

File: src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts (hunk 1 above)
Comment (michael-wirth) [reply src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts]: @Invizible please rename the suffix `El` to `ElementRef` - it's longer but I think its clearer

**Pull request merged**